### PR TITLE
osd/OSDCap: rbd profile permits use of rbd_info

### PR DESCRIPTION
**Pull request by Songweibin**
Fixes: http://tracker.ceph.com/issues/39973

Signed-off-by: songweibin <song.weibin@zte.com.cn>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/osd/OSDCap.cc [hunk 1]
```diff
@@ -336,6 +336,8 @@ void OSDCapGrant::expand_profile()
                                 OSDCapSpec(osd_rwxa_t(OSD_CAP_CLS_R)));
     profile_grants.emplace_back(OSDCapMatch(string(), "rbd_mirroring"),
                                 OSDCapSpec(osd_rwxa_t(OSD_CAP_CLS_R)));
+    profile_grants.emplace_back(OSDCapMatch(string(), "rbd_info"),
+                                OSDCapSpec(osd_rwxa_t(OSD_CAP_CLS_R)));
```
Comment (dillaman) [inline src/osd/OSDCap.cc]: We actually just want to restrict to invoking "rbd.metadata_list" cls method (see https://github.com/ceph/ceph/pull/28253/files#diff-1ad4853f970880c78ea0e52c81e621b4L351 below for an example of how to restrict by class method)

Comment (dillaman) [reply src/osd/OSDCap.cc]: ... and the ```OSDCapMatch(string(), ...``` should copy the pool restriction (if any) from the profile cap so it it doesn't unnecessarily give global access to that object.

Comment (Songweibin) [reply src/osd/OSDCap.cc]: > ... and the `OSDCapMatch(string(), ...` should copy the pool restriction (if any) from the profile cap so it it doesn't unnecessarily give global access to that object.

@dillaman  Updated, I am not sure if I got your point of "copy the pool restriction (if any) from the profile cap". Thank you.

Comment (dillaman) [reply src/osd/OSDCap.cc]: See [1] where the pool/namespace restriction is applied to the ```OSDCapMatch```. The difference is that we would only want to include the pool restriction (not the ns restriction), so you could pass ```OSDCapMatch(profile.pool_namespace.pool_name)```

[1] https://github.com/ceph/ceph/pull/28253/files#diff-1ad4853f970880c78ea0e52c81e621b4R348

**Comment by Songweibin** [comment]
@dillaman  updated, thank you :-)

**Pull request merged**